subprojects/signing/src/main/java/org/gradle/plugins/signing/signatory/pgp/PgpSignatoryFactory.java

```diff
@@ -81,8 +81,16 @@ protected PgpSignatory readProperties(Project project, String prefix, String nam
         for (String property : PROPERTIES) {
             String qualifiedProperty = (String)getQualifiedPropertyName(prefix, property);
             if (project.hasProperty(qualifiedProperty)) {
-                values.add(
-                    project.property(qualifiedProperty));
+                Object prop = project.property(qualifiedProperty);
+                if(prop != null){
+                    values.add(prop);
+                } else {
+                    if(required){
```

Comment:
The exception messages differ. Another concern was keeping the C/M/L on the exceptions thrown therein as close to the source of the problem as possible. If I extract this to a method like this:

```groovy
private boolean throwIfRequired(boolean required, String message)
```
then I'll have to check the return value of the method in order to know if I should return early from the `readProperties` method. 

I'm drawing a blank presently for ways to accomplish this more cleanly in Groovy.